### PR TITLE
feat: CLI execution tests POC

### PR DESCRIPTION
**Pull request by lukas-valenta**
## Description

New way of testing - by execution of the CLI itself in temporary directory

## Motivation and Context

It was difficult to test stuff that touches filesystem and accesses globals like process.cwd() in isolation.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: fixtures/profiles/starwars/character-information@1.0.2.supr [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name = "starwars/character-information"
+version = "1.0.1"
```
Comment (lukas-valenta) [inline fixtures/profiles/starwars/character-information@1.0.2.supr:2]: shouldn't this say 1.0.2? :trollface:

File: fixtures/profiles/starwars/character-information@1.0.2.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"profile_id":"starwars/character-information@1.0.1","profile_name":"starwars/character-information","profile_version":"1.0.1","url":"https://superface.dev/starwars/character-information@1.0.1","owner":"hello","owner_url":"","published_at":"2021-05-17T08:55:54.905Z","published_by":"hello <hello@superface.ai>"}
```
Comment (lukas-valenta) [inline fixtures/profiles/starwars/character-information@1.0.2.json:1]: shouldn't this say 1.0.2? :trollface:

File: fixtures/profiles/starwars/character-information@1.0.2.supr.ast.json [hunk 3]
```diff
@@ -0,0 +1 @@
...art":463,"end":469},"location":{"line":33,"column":14}},"location":{"line":29,"column":1},"span":{"start":411,"end":469},"title":"Height","description":"The height of the character"},{"kind":"NamedFieldDefinition","fieldName":"weight","type":{"kind":"PrimitiveTypeName","name":"string","span":{"start":523,"end":529},"location":{"line":39,"column":14}},"location":{"line":35,"column":1},"span":{"start":471,"end":529},"title":"Weight","description":"The weight of the character"},{"kind":"NamedFieldDefinition","fieldName":"yearOfBirth","type":{"kind":"PrimitiveTypeName","name":"string","span":{"start":602,"end":608},"location":{"line":45,"column":19}},"location":{"line":41,"column":1},"span":{"start":531,"end":608},"title":"Year of birth","description":"The year of birth of the character"},{"kind":"NamedFieldDefinition","fieldName":"message","type":{"kind":"PrimitiveTypeName","name":"string","span":{"start":706,"end":712},"location":{"line":51,"column":15}},"location":{"line":47,"column":1},"span":{"start":610,"end":712},"title":"Message","description":"The message for when an error occurs looking up character information"}],"location":{"line":1,"column":1},"span":{"start":0,"end":712}}
```
Comment (lukas-valenta) [inline fixtures/profiles/starwars/character-information@1.0.2.supr.ast.json:1]: shouldn't this say 1.0.2? :trollface:

File: src/test/utils.ts [hunk 4]
```diff
@@ -0,0 +1,171 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable no-async-promise-executor */
```
Comment (lukas-valenta) [inline src/test/utils.ts:2]: i'm not sure i like what's going on in this file :sweat_smile:

File: src/test/utils.ts [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+    cwd: directory,
+    env: { ...process.env, ...env, SUPERFACE_API_URL: apiUrl },
+  });
+
+  childProcess.stdin?.setDefaultEncoding('utf-8');
+
+  let currentInputTimeout: NodeJS.Timeout, killIOTimeout: NodeJS.Timeout;
+
+  // Creates a loop to feed user inputs to the child process in order to get results from the tool
+  const loop = (inputs: string[]) => {
```
Comment (lukas-valenta) [inline src/test/utils.ts:106]: don't shadow names pls. add rule to eslint pls, it's not on for some reason: https://eslint.org/docs/rules/no-shadow

File: src/test/utils.ts [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    if (killIOTimeout) {
+      clearTimeout(killIOTimeout);
+    }
+
+    if (!inputs.length) {
+      childProcess.stdin?.end();
+
+      // Set a timeout to wait for CLI response. If CLI takes longer than
+      // maxTimeout to respond, kill the childProcess and notify user
+      killIOTimeout = setTimeout(() => {
```
Comment (lukas-valenta) [inline src/test/utils.ts:116]: do you really want to set this in each loop?

File: src/test/utils.ts [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+ * @export
+ * @param {string} directory - the directory in which the process runs
+ * @param {string[]} args - arguments of the process
+ * @param {string} apiUrl - the API URL (to be overriden with mock)
+ * @param {string [] | undefined} inputs - the inquier prompt inputs
+ * @param {boolean} debug - pass child process stdout to console
+ * @param {NodeJS.ProcessEnv} [env] - any additional environment variables
+ * @returns  {Promise<string>} - result is concatenated stdout
+ */
+export async function execCLI(
```
Comment (lukas-valenta) [inline src/test/utils.ts:83]: i'd go with object at this point

File: src/test/utils.ts (hunk 6 above)
Comment (Jakub-Vacek) [reply src/test/utils.ts:116]: only if `(!inputs.length) {`

**Pull request merged**